c++/src/capnp/compiler/generics.c++

```diff
@@ -131,10 +131,24 @@ bool BrandedDecl::compileAsType(
         }
 
         if (elementType.isAnyPointer()) {
-          addError(errorReporter, "'List(AnyPointer)' is not supported.");
-          // Seeing List(AnyPointer) later can mess things up, so change the type to Void.
-          elementType.setVoid();
-          return false;
+          auto unconstrained = elementType.getAnyPointer().getUnconstrained();
+
+          if (unconstrained.isAnyKind()) {
+            addError(errorReporter, "'List(AnyPointer)' is not supported.");
+            // Seeing List(AnyKind) later can mess things up, so change the type to Void.
+            elementType.setVoid();
+            return false;
+          } else if (unconstrained.isStruct()) {
+            addError(errorReporter, "'List(AnyStruct)' is not supported.");
+            // Seeing List(Struct) later can mess things up, so change the type to Void.
+            elementType.setVoid();
+            return false;
+          } else if (unconstrained.isList()) { // TODO This should be supported
```

Comment:
I think I'd leave this as supported in the compiler for now, even though the C++ library currently mishandles it. Other implementations might handle it just fine.

In `test.capnp` I'd add the `List(AnyList)` line but comment it out and put `# TODO(soon): Make List(AnyList) work correctly in C++ generated code.`